kedro/framework/cli/pipeline.py

```diff
@@ -876,7 +876,7 @@ def _generate_wheel_file(
     pipeline_name: str,
     destination: Path,
     source_paths: Tuple[_SourcePathType, ...],
-    version: str,
+    version: Optional[str],
```

Comment:
I don't understand this change and the one to the signature of `_generate_setup_file`. When I run `mypy` on this file I don't get any errors here, and it looks like you're changing it to put an optional argument in front of a compulsory one which shouldn't be allowed?
```
    def f(a=1, b): pass
         ^
SyntaxError: non-default argument follows default argument
```

Reply:
I don't think this is correct, because on L697 we make sure we don't pass `None` around

```python
version = version or _get_default_version(metadata, pipeline_name)
```

So the type hint should be `str` only.

As for
> it looks like you're changing it to put an optional argument in front of a compulsory one which shouldn't be allowed?

`Optional[str]` is just the equivalent of `Union[str, None]`. It's not the same as an optional argument, i.e. kwargs. That's denoted by giving it a default value, like we do 2 lines below `alias: str = None`, rather than by the type hint.

Reply:
Not sure what I was thinking on this one, looking at the commit message, it looks geared towards the line below that is `alias: str = None`